modules/gapi/src/api/gproto.cpp

```diff
@@ -197,6 +197,20 @@ bool cv::can_describe(const GMetaArgs &metas, const GRunArgs &args)
                      });
 }
 
+bool cv::nonzero_dims_in_inmat(const GMetaArgs& metas)
+{
+    bool state = true;
+    for (const auto& current_meta : metas)
+    {
+        if (current_meta.index() == cv::GMetaArg::index_of<cv::GMatDesc>())
+        {
+            const auto& desc_mat = cv::util::get<cv::GMatDesc>(current_meta);
+            state &= (desc_mat.size.height != 0 && desc_mat.size.width != 0);
+        }
+    }
+    return state;
```

Comment:
There's no need to check every argument, you can return `false` early once you find a match.

ALSO: `std::any_of`

Reply:
Also I'd refactor the code in other form. Lets have a function like `validate_input_arg` which itself takes a `GRunArg` and checks if the parameter is valid based on its type. If anything wrong for, say, Mat, you just throw an exception.

Reply:
And yes, please start with `GRunArgs` not `GMetaArgs`.

Reply:
Fixed. Another form of validation. Is it?